homeassistant/components/statistics/sensor.py

```diff
@@ -885,13 +899,19 @@ def _stat_value_min(self) -> StateType:
         return None
 
     def _stat_variance(self) -> StateType:
+        if len(self.states) == 1:
+            return 0.0
         if len(self.states) >= 2:
             return statistics.variance(self.states)
         return None
 
     # Statistics for binary sensor
 
     def _stat_binary_average_step(self) -> StateType:
+        if len(self.states) == 1:
+            if self.states[0] is True:
+                return 100.0
+            return 0.0
```

Comment:
The codecov CI job says there's lack of code coverage.

Please confirm this is the case by running coverage locally and add a test if it is.
```
pytest --cov=homeassistant.components.statistics --cov-report html tests/components/statistics/
```

Reply:
Code coverage was improved now. There remains one uncovered line, but that has nothing to do with this PR